### PR TITLE
Add notes on our coding style

### LINKED ISSUE #533: Articulate the coding style used in the standard
Several PRs updated the coding style for the examples in the standard. We need to update the admin guide to include a discussion of what our expectations are for code

PRs to review for style notes:

- #469
- #470 
- #471 
- #474
- #479 
- #483 
- #485 
- #487 
- #488 
- #500

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #533

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -89,3 +89,27 @@ The tool will replace the text with the following:
...
+- Use a blank line between peer declarations. There is two exceptions to this rule:
+  - If a group of peer declarations all have empty bodies (e.g. `class C {}` or `public void M() {}`, then no blank line is added between them.
+  - Don't add a blank line where the concept is better illustrated without them. For example, multiple overloads of the same method may read better grouped without a blank line.
+- Elements of the same type can be grouped without blank lines. When that's done, there should be a blank line between elements of different types. (e.g. all private fields can be declared without a blank line, but there should be a blank line between the group of private fields and method declarations).
+- Indentation is 4 spaces. Use spaces, not tabs for the Word and PDF export formats.
+- Where a group of lines all have comments appended, the opening `//` should be on the same column on all lines.
+- Where expression bodied members are used, place all code on the same line if it fits. If it's too long, wrap after the `=>` token.
+- Line lengths cannot exceed 80 characters.
```
Comment (BillWagner) [inline CONTRIBUTING.md:108]: Jon's earlier PR for line length used 95 characters. The word converter has a warning at 80 characters, so I used that instead.

**Comment by RexJaeschke** [comment]
@BillWagner I tried to approve this (without comment), but when I selected "Add you review" it comes up with a "New Issue" button and no way to approve!

File: CONTRIBUTING.md (hunk 1 above)
Comment (jskeet) [reply CONTRIBUTING.md:108]: Yup - I had to change from 95 to 80 due to a change in the Word font.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -89,3 +89,27 @@ The tool will replace the text with the following:
...
+
+## Coding style
+
+The code in the standard is optimized for reading the standard. In general, follow the existing code format. We've tried to be consistent throughout the standard.
+
+We've adopted the following non-exhaustive rules:
+
+- Use the latest C# syntax for the branch being edited. This is currently C# 7.3, except where older syntax is required for the example. Readers of the standard should see the latest syntax.
+- Use [Allman style braces](https://en.wikipedia.org/wiki/Indentation_style?msclkid=5bd8d8f5cd7b11ec82461327ced4546a#Allman_style) in most of our samples. The exception is when a method or type is empty, where the declaration, the opening and closing brace are on the same line.
+- Use a blank line between peer declarations. There is two exceptions to this rule:
```
Comment (jskeet) [inline CONTRIBUTING.md]: is => are

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -89,3 +89,27 @@ The tool will replace the text with the following:
...
+- Elements of the same type can be grouped without blank lines. When that's done, there should be a blank line between elements of different types. (e.g. all private fields can be declared without a blank line, but there should be a blank line between the group of private fields and method declarations).
+- Indentation is 4 spaces. Use spaces, not tabs for the Word and PDF export formats.
+- Where a group of lines all have comments appended, the opening `//` should be on the same column on all lines.
+- Where expression bodied members are used, place all code on the same line if it fits. If it's too long, wrap after the `=>` token.
+- Line lengths cannot exceed 80 characters.
+- Use spaces around most tokens (e.g. `:` for base classes or interface implementation, `+` and all operators). Exceptions are:
+  - `<` and `>` to declare generic types and methods
+  - `(` and `)`. No spaces around parentheses on method declarations.
+  - No space after the opening `(` and no space preceding the closing `)` for other expressions.
+  - no spaces on `[]` for array declarations.
```
Comment (jskeet) [inline CONTRIBUTING.md]: "no" => "No"

**Review by jskeet — APPROVED** [review]
A couple of nits, but that's all.

File: CONTRIBUTING.md (hunk 2 above)
Comment (BillWagner) [inline CONTRIBUTING.md]:
```suggestion
- Use a blank line between peer declarations. There are two exceptions to this rule:
```

File: CONTRIBUTING.md (hunk 3 above)
Comment (BillWagner) [inline CONTRIBUTING.md]:
```suggestion
  - No spaces on `[]` for array declarations.
```

**Pull request merged**